lib/main.dart

```diff
@@ -61,14 +67,128 @@ const kBGPushTimeout = Duration(seconds: 28);
+      if (collectionID == null) {
+        await hw.HomeWidget.saveWidgetData(
+          "slideshow",
+          null,
+        );
+
+        await hw.HomeWidget.updateWidget(
+          name: 'SlideshowWidgetProvider',
+          androidName: 'SlideshowWidgetProvider',
+          qualifiedAndroidName: 'io.ente.photos.SlideshowWidgetProvider',
+          iOSName: 'SlideshowWidget',
+        );
+        return false;
+      }
+
+      try {
+        await hw.HomeWidget.setAppGroupId(iOSGroupID);
+        final res = await FilesDB.instance.getFilesInCollection(
+          collectionID,
+          galleryLoadStartTime,
+          galleryLoadEndTime,
+        );
+
+        final previousGeneratedId =
+            await hw.HomeWidget.getWidgetData<int>("home_widget_last_img");
+        final files = res.files
+            .where((element) => element.generatedID != previousGeneratedId);
+        final randomNumber = Random().nextInt(files.length);
+        final randomFile = files.elementAt(randomNumber);
+        final fullImage = await getFileFromServer(randomFile);
+        if (fullImage == null) return false;
+
+        Image img = Image.file(fullImage);
+        var imgProvider = img.image;
+        await PreloadImage.loadImage(imgProvider);
+
+        img = Image.file(fullImage);
+        imgProvider = img.image;
+
+        final image = await decodeImageFromList(await fullImage.readAsBytes());
+        final width = image.width.toDouble();
+        final height = image.height.toDouble();
+        final size = min(width, height);
+
+        final widget = ClipRRect(
+          borderRadius: BorderRadius.circular(32),
+          child: Container(
+            width: size,
+            height: size,
+            decoration: BoxDecoration(
+              color: Colors.black,
+              image: DecorationImage(image: imgProvider, fit: BoxFit.cover),
+            ),
+          ),
+        );
+
+        await hw.HomeWidget.renderFlutterWidget(
+          widget,
+          logicalSize: Size(size, size),
+          key: "slideshow",
+        );
+
+        await hw.HomeWidget.updateWidget(
+          name: 'SlideshowWidgetProvider',
+          androidName: 'SlideshowWidgetProvider',
+          qualifiedAndroidName: 'io.ente.photos.SlideshowWidgetProvider',
+          iOSName: 'SlideshowWidget',
+        );
+
+        if (randomFile.generatedID != null) {
+          await hw.HomeWidget.saveWidgetData<int>(
+            "home_widget_last_img",
+            randomFile.generatedID!,
+          );
+        }
+
+        _logger.info(
+          ">>> SlideshowWidget rendered with size ${width}x$height",
+        );
+        return true;
+      } catch (_) {
+        return false;
+      }
+    },
+  );
+}
+
 void main() async {
   debugRepaintRainbowEnabled = false;
   WidgetsFlutterBinding.ensureInitialized();
   MediaKit.ensureInitialized();
+
+  if (Platform.isAndroid) {
+    try {
+      await Workmanager()
```

Comment:
Let's not await on this in `main()` and instead use `.then()` and `.onError()` on the futures? Don't want to increase the time app stays on splash screen.

Reply:
Done! Although there is analysis check that requires await so I have used unawaited to fix this.